src/fhir_search/uri_parser.clj

```diff
@@ -0,0 +1,140 @@
+(ns fhir-search.uri-parser
+  (:require [clojure.string :as str])
+  (:require [clojure.walk :refer [postwalk]])
+  (:import [java.net URI]))
+
+(def prefixes ["eq" "ne" "gt" "lt" "ge" "le" "sa" "eb" "ap"])
+
+(defn prefix-find "Find the prefix of a number value. If there are more than two prefixes or the value is not the correct returns nil."
+  [pf-list element]
+  (when-let [prefix (first (filter (and #(re-find (re-pattern %) element)
+                                        #(not (re-find #"\p{L}" (str/replace element (re-pattern %) "")))) pf-list))]
+    prefix))
+
+
+(defn url-encoding [url]
```

Comment:
I don't get the need for this fn? Are you demonstrating that java.net.URI already decodes the query-string when you call de `(.getQuery uri)` method? If so, that's great news, and lets remove this fn at all if we don't need it.

Reply:
oh! I see, great!

```clj
  (.getQuery (URI. "/Observation?code=http%3A%2F%2Floinc.org%7C8867-4&value-quantity=lt60%2Cgt100"))
  ;;=> "code=http://loinc.org|8867-4&value-quantity=lt60,gt100"
  ```